src/plugins/console/public/application/containers/editor/monaco/utils/tokens_utils.ts

```diff
@@ -0,0 +1,480 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import {
+  ampersandRegex,
+  digitRegex,
+  equalsSignRegex,
+  newLineRegex,
+  numberStartRegex,
+  questionMarkRegex,
+  slashRegex,
+  whitespacesRegex,
+} from './constants';
+
+/*
+ * This function parses a line with the method and url.
+ * The url is parsed into path and params, each parsed into tokens.
+ * Returns method, urlPathTokens and urlParamsTokens which are arrays of strings.
+ */
+export const parseLine = (line: string): ParsedLineTokens => {
+  // try to parse into method and url (split on whitespace)
+  const parts = line.split(whitespacesRegex);
+  // 1st part is the method
+  const method = parts[0];
+  // 2nd part is the url
+  const url = parts[1];
+  // try to parse into url path and url params (split on question mark)
+  const { urlPathTokens, urlParamsTokens } = parseUrl(url);
+  return { method, urlPathTokens, urlParamsTokens };
+};
+
+/*
+ * This function parses an url into path and params, each parsed into tokens.
+ * Returns urlPathTokens and urlParamsTokens which are arrays of strings.
+ */
+export const parseUrl = (
+  url: string
+): {
+  urlPathTokens: ParsedLineTokens['urlPathTokens'];
+  urlParamsTokens: ParsedLineTokens['urlParamsTokens'];
+} => {
+  let urlPathTokens: ParsedLineTokens['urlPathTokens'] = [];
+  let urlParamsTokens: ParsedLineTokens['urlParamsTokens'] = [];
+  const urlParts = url.split(questionMarkRegex);
+  // 1st part is the url path
+  const urlPath = urlParts[0];
+  // try to parse into url path tokens (split on slash)
+  if (urlPath) {
+    urlPathTokens = urlPath.split(slashRegex);
+  }
+  // 2nd part is the url params
+  const urlParams = urlParts[1];
+  // try to parse into url param tokens
+  if (urlParams) {
+    urlParamsTokens = urlParams.split(ampersandRegex).map((urlParamsPart) => {
+      return urlParamsPart.split(equalsSignRegex);
+    });
+  }
+  return { urlPathTokens, urlParamsTokens };
+};
+
+/*
+ * This function parses the body of the request into tokens.
+ * For example '{ "test": [' -> ['{', 'test', '[']. This array is used for autocomplete.
+ * Returns array of strings representing body tokens for autocomplete.
+ */
+export const parseBody = (value: string): string[] => {
```

Comment:
This function uses the logic similar to the code in parser, we could consider re-using some of the code in both.